src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -215,6 +215,7 @@ public class LockFileBuilder
                         var package = packageInfo.Package;
 
                         var targetLibrary = LockFileUtils.CreateLockFileTargetLibrary(
+                            tfi.Dependencies.FirstOrDefault(e => e.Name.Equals(library.Name, StringComparison.OrdinalIgnoreCase)),
```

Comment:
I see 2 instances where Dependencies are searching for the same where clause.  I think we can store the result into a variable so that it can be passed as argument while invoking `LockFileUtils.CreateLockFileTargetLibrary` method.

Reply:
fixed.